src/main/java/org/apache/sling/commons/threads/impl/ThreadPoolMetricsGauges.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.sling.commons.threads.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.sling.commons.metrics.Gauge;
+import org.apache.sling.commons.threads.jmx.ThreadPoolMBean;
+
+class ThreadPoolMetricsGauges {
+
+    public static Map<String, Gauge<?>> create(final ThreadPoolMBean bean) {
```

Comment:
Have you considered reading the JMX metadata instead of harcoding the metric names?

Reply:
@rombert I could try with https://docs.oracle.com/javase/7/docs/api/javax/management/MBeanAttributeInfo.html#getType() (especially as [gauge](https://github.com/apache/sling-org-apache-sling-commons-metrics/blob/master/src/main/java/org/apache/sling/commons/metrics/Gauge.java ) instances are parametrized) - I just need to exclude MaxThreadAge as it's deprecated, returns constant -1, so there is no use to make it a gauge.

Reply:
@rombert Should it look instead similar to the one below?
```
  public static Map<String, Gauge<?>> create(final DynamicMBean bean, MBeanInfo info) {
        Map<String, Gauge<?>> gauges = new HashMap<>();
        for(final MBeanAttributeInfo attr : info.getAttributes()) {
            
            if(attr.getType() == "int" || attr.getType() == "java.lang.Integer") {
            
                gauges.put(attr.getName(), new Gauge<Integer>() {
                    public Integer getValue() {
                        try {
                            return (Integer)bean.getAttribute(attr.getName());
                        } catch (JMException e) {
                            //???
                            return -1;
                        }
                    }
                });
            
            }
            
            // and for other possible types???
            
        }
        return gauges;
    }
```

Reply:
Yes, I guess that API is a good start.

Reply:
@rombert hopefully https://github.com/apache/sling-org-apache-sling-commons-threads/pull/1/commits/2680f32209758f63c895194e6081706ba6c486de resolves it